### DISCUSSION TITLE
Fixed bug: Double colon pathname will cause assertion failure

### DISCUSSION THREAD

**Discussion by ecm-pushbx** [dosemu2/dosemu2#2115] [General]
Running `dosemu -t -K "$PWD"` then running `test` in DOS would lead to an assertion failure in earlier dosemu2 revisions:

```
Welcome to dosemu2!
    Build 2.0pre9-dev-20231001-1576-g9c5924ec0
E:\>testERROR: KVM: error opening /dev/kvm: No such file or directory
snd-virmidi module not loaded or device "default" not configured
see "amidi -l" for the list of midi devices
ERROR: MIDI: unsupported synth mode mt32
ERROR: Unable to open console or check with X to evaluate the keyboard map.
Please specify your keyboard map explicitly via the $_layout option. Defaulting to
US.
dosemu.bin: lfn.c:492: build_truename: Assertion `dd >= 0' failed.

ERROR: cpu exception in dosemu code outside of VM86()!
Aborted (Signal sent by tkill() 225524 1001)
```

Here's the contents of the test file's assembly sources, assemble using `nasm test.asm -o test.com`

```
	cpu 8086
	org 256

start:
	mov ax, 716Ch
	xor di, di
	mov si, name
	mov bx, 6020h
	mov dx, 1
	mov cx, 0
	stc
	int 21h
	mov ax, 4C00h
	int 21h

name:
	db "::script::list.eld", 0
```

This seems to be fixed by https://github.com/dosemu2/dosemu2/commit/13f4867c2c90d15a8cacdb8196dda45b005d61b8

**Comment by stsp** [comment]
What does it have to do with -K
or anything alike?
You just passed a bad name to
716ch, and that's all.

**Reply by ecm-pushbx** [reply]
The `-K` is just to set up the test. And yes, it is a bad name, but DOS or mfs or redirectors generally shouldn't crash on a bad name, and rather return the error to the caller. Which it does now.

I just wanted to document that a double-colon name used to cause dosemu2 to crash, due to the bug you fixed in that revision.
